packages/twenty-front/src/modules/workflow/components/WorkflowDiagramStepNode.tsx

```diff
@@ -15,12 +21,26 @@ const StyledStepNodeLabelIconContainer = styled.div`
 `;
 
 export const WorkflowDiagramStepNode = ({
+  id,
   data,
+  selected,
 }: {
+  id: string;
   data: WorkflowDiagramStepNodeData;
+  selected?: boolean;
 }) => {
   const theme = useTheme();
 
+  const workflowId = useRecoilValue(workflowIdState);
+
+  const workflowWithCurrentVersion = useWorkflowWithCurrentVersion(workflowId);
+  assertWorkflowWithCurrentVersionIsDefined(workflowWithCurrentVersion);
+
+  const { deleteOneStep } = useDeleteOneStep({
+    workflow: workflowWithCurrentVersion,
+    stepId: id,
+  });
```

Comment:
I don't feel it is good to subscribe to workflow changes in every Reactflow node. I only need to know what are the current trigger and steps in the `deleteOneStep` function, so I think it might be better to get the data from the cache at this moment.